### PR TITLE
Add whatsnew for arrow

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

cc @mroeschke

I could add my blogpost https://medium.com/p/2891d3d96d2b as evidence for the performance improvements

### CODE REVIEW

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -14,6 +14,46 @@ including other versions of pandas.
 Enhancements
 ~~~~~~~~~~~~
 
+.. _whatsnew_210.enhancements.pyarrow_dependency:
+
+PyArrow will become a required dependency with pandas 3.0
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+[PyArrow](https://arrow.apache.org/docs/python/index.html) will become a required
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:22]:
```suggestion
`PyArrow <https://arrow.apache.org/docs/python/index.html>`_ will become a required
```

Unfortunately still not using markdown :(

File: doc/source/whatsnew/v2.1.0.rst [hunk 2]
```diff
@@ -14,6 +14,46 @@ including other versions of pandas.
...
+strings as PyArrow backed strings with dtype ``pd.ArrowDtype(pa.string())`` instead.
+This option only works if PyArrow is installed. PyArrow backed strings have a
+significantly reduced memory footprint and provide a big performance improvement
+compared to NumPy object.
+
+The option can be enabled with:
+
+```
+pd.options.future.infer_string = True
+```
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:53]:
```suggestion
.. code-block:: python

    pd.options.future.infer_string = True
```
```

File: doc/source/whatsnew/v2.1.0.rst [hunk 3]
```diff
@@ -14,6 +14,46 @@ including other versions of pandas.
...
+
+It is expected that this behavior will become the default with pandas 3.0.
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:55]:
```suggestion
This behavior will become the default with pandas 3.0.
```

File: doc/source/whatsnew/v2.1.0.rst [hunk 4]
```diff
@@ -14,6 +14,46 @@ including other versions of pandas.
...
 ~~~~~~~~~~~~
 
+.. _whatsnew_210.enhancements.pyarrow_dependency:
+
+PyArrow will become a required dependency with pandas 3.0
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`PyArrow <https://arrow.apache.org/docs/python/index.html>`_ will become a required
+dependency of pandas starting with pandas 3.0. This decision was made based on
+[PDEP 12](https://pandas.pydata.org/pdeps/0010-required-pyarrow-dependency.html).
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:24]:
```suggestion
`PDEP 12 <https://pandas.pydata.org/pdeps/0010-required-pyarrow-dependency.html>`_.
```

File: doc/source/whatsnew/v2.1.0.rst [hunk 5]
```diff
@@ -14,6 +14,46 @@ including other versions of pandas.
...
+This will enable more changes that are hugely beneficial to pandas users, including
+but not limited to:
+
+- inferring strings as PyArrow backed strings by default enabling a significant
+  reduction of the memory footprint and huge performance improvements.
+- inferring more complex dtypes with PyArrow by default, like ``Decimal``, ``lists``,
+  ``bytes``, ``structured data`` and more.
+- Better interoperability with other libraries that depend on Apache Arrow.
+
+We are collecting feedback on this decision [here](https://github.com/pandas-dev/pandas/issues/54466).
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:35]:
```suggestion
We are collecting feedback on this decision `here <https://github.com/pandas-dev/pandas/issues/54466>`_.
```

**Comment by phofl** [comment]
Sorry, wrote too many blogs...

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**